app/src/main/java/io/gnosis/kouban/ui/address/capture/AddressCaptureFragment.kt

```diff
@@ -50,7 +50,10 @@ class AddressCaptureFragment : BaseFragment<FragmentAddressCaptureBinding>() {
         super.onActivityResult(requestCode, resultCode, data)
         if (requestCode == QR_SCAN_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
             QRCodeScanActivity.handleResult(QR_SCAN_REQUEST_CODE, resultCode, data) { safeAddressString ->
-                viewModel.handleSafeAddress(safeAddressString.asEthereumAddress()!!)
+                safeAddressString.asEthereumAddress()?.let { safeAddress ->
+                    binding.addressInput.updateAddress(safeAddress)
```

Comment:
we should set addressInput only after it was handled/verified. I think SafeAddressUpdated -> .. in consumeViewStates() is a better place for it.

Reply:
If you look at the other change in this commit, that's exactly where this was set and was causing the issues.